### ISSUE TITLE
Function parameters hinting doesn't work inside other functions.

### ISSUE THREAD

**Issue by phil-harmoniq** [python/mypy#3896]
Example one:

    def add(thing: int, other_thing: int) -> int:
        return thing + other_thing
    
    add("string", "other string")

This will cause mypy to give me a warning saying I'm using incompatible types when calling `add()`. However, if I change the code to:

    def add(thing: int, other_thing: int) -> int:
        return thing + other_thing
    
    def main():
        add("string", "other string")
    
    main()

mypy doesn't see a problem with this at all. I hope I'm not missing something. Let me know if you need any other information.

**Comment by ethanhs** [comment]
This is a subtle thing mypy does. Mypy doesn't check untyped (un-annotated) functions. So your main function, nor its contents are checked. If you modify main to be 

```python

def main() -> None:
    add("string", "other string")
```
Mypy should report the error. Additionally you can add the flag `--check-untyped-defs` and mypy will check untyped functions (such as your current main function). Hope that helps! Feel free to leave a comment or drop by our [gitter chat](https://gitter.im/python/typing) if you have more questions.

**Comment by phil-harmoniq** [comment]
Thanks for the quick response! That would explain why my problem was so hard to find Google haha.
